File: leetcode-blind75/twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let hashMap = {};
    let result = [];
    
    for (let i = 0; i < nums.length; i++) {
        let difference = target - nums[i];
        
        if (difference in hashMap) {
            result.push(hashMap[difference], i);
        }
        
        hashMap[nums[i]] = i;
    }
    return result;
};

File: leetcode-blind75/twoSum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        int len = nums.Length;
        //declaring a new dictionary
        Dictionary<int, int> numsDict = new();
        
        for (int i = 0; i < len; i++)
        {
            int diff = target - nums[i];
            if (numsDict.ContainsKey(diff))
            {
                return new int[]{numsDict[diff], i};
            }
            else
            {
                if(!numsDict.ContainsKey(nums[i]))
                {
                    numsDict.Add(nums[i],i);
                }
            }
        }
        return new int[0];
    }
}